contracts/vaults/STETHVault/RibbonThetaSTETHVault.sol

```diff
@@ -0,0 +1,358 @@
+                USDC: USDC,
+                currentOption: oldOption,
+                delay: delay,
+                lastStrikeOverride: lastStrikeOverride,
+                overriddenStrikePrice: overriddenStrikePrice
+            });
+
+        (address otokenAddress, uint256 premium, , ) =
+            VaultLifecycleSTETH.commitAndClose(
+                strikeSelection,
+                optionsPremiumPricer,
+                premiumDiscount,
+                closeParams,
+                vaultParams,
+                vaultState,
+                address(collateralToken)
+            );
+
+        currentOtokenPremium = uint104(premium);
+        optionState.nextOption = otokenAddress;
+        optionState.nextOptionReadyAt = uint32(block.timestamp.add(delay));
+
+        _closeShort(oldOption);
+    }
+
+    /**
+     * @notice Closes the existing short position for the vault.
+     */
+    function _closeShort(address oldOption) private {
+        optionState.currentOption = address(0);
+
+        uint256 lockedAmount = vaultState.lockedAmount;
+        vaultState.lastLockedAmount = uint104(lockedAmount);
+        vaultState.lockedAmount = 0;
+
+        if (oldOption != address(0)) {
+            uint256 withdrawAmount =
+                VaultLifecycleSTETH.settleShort(GAMMA_CONTROLLER);
+            emit CloseShort(oldOption, withdrawAmount, msg.sender);
+        }
+    }
+
+    /**
+     * @notice Rolls the vault's funds into a new short position.
+     */
+    function rollToNextOption() external onlyKeeper nonReentrant {
+        (address newOption, uint256 queuedWithdrawAmount) = _rollToNextOption();
+
+        // Locked balance denominated in `collateralToken`
+
+        uint256 lockedBalance =
+            collateralToken.balanceOf(address(this)).sub(
+                collateralToken.getWstETHByStETH(queuedWithdrawAmount)
+            );
+
+        emit OpenShort(newOption, lockedBalance, msg.sender);
+
+        VaultLifecycleSTETH.createShort(
+            GAMMA_CONTROLLER,
+            MARGIN_POOL,
+            newOption,
+            lockedBalance
+        );
+
+        _startAuction();
+    }
+
+    /**
+     * @notice Initiate the gnosis auction.
+     */
+    function startAuction() external onlyKeeper nonReentrant {
+        _startAuction();
+    }
+
+    function _startAuction() private {
+        GnosisAuction.AuctionDetails memory auctionDetails;
+
+        uint256 currOtokenPremium = currentOtokenPremium;
+
+        require(currOtokenPremium > 0, "!currentOtokenPremium");
+
+        auctionDetails.oTokenAddress = optionState.currentOption;
+        auctionDetails.gnosisEasyAuction = GNOSIS_EASY_AUCTION;
+        auctionDetails.asset = vaultParams.asset;
+        auctionDetails.assetDecimals = vaultParams.decimals;
+        auctionDetails.oTokenPremium = currOtokenPremium;
+        auctionDetails.duration = auctionDuration;
+
+        optionAuctionID = VaultLifecycle.startAuction(auctionDetails);
```

Comment:
not sure if best to disregard this:

At the end of the originally called startAuction function, the optionAuctionID state variable is
modified, violating the recommended Check-Effects-Interaction pattern.